### DISCUSSION TITLE
Can I specify system properties when installing app?

### DISCUSSION THREAD

**Discussion by tadayosi** [jbangdev/jbang#1736] [Q&A] [answered]
Hi,

When we install an app from a remote catalog like:
```console
jbang app install hawtio@hawtio/hawtio
```
it will create a local script such as:
```shell
$ cat .jbang/bin/hawtio
#!/bin/sh
exec jbang run hawtio@hawtio/hawtio "$@"
```
but sometimes we might want to specify some system properties to customise the installed app. So far it appears we can only do so by manually modifying the generated script:
```shell
#!/bin/sh
exec jbang run -Dhawtio.jbang.version=4.0.0.redhat-00029 hawtio@hawtio/hawtio "$@"
```

Is there a smarter way to achieve the same thing?  If there isn't, it might be a good feature enhancement for `jbang app install`.

Thank you!

**Comment by quintesse** [comment, marked as answer]
Hi @tadayosi , you're right that at this moment this is not possible and it would indeed be a good idea for an enhancement. I think the only reason we haven't done so yet is because it wasn't immediately clear what would be the best way to add this (not technically but what would be the best UX for the install command). So if you have any good ideas we'd be glad to hear them :-)

**Reply by tadayosi** [reply]
Thanks for your confirmation. So a straighforward idea for the enhancement would look like:
```
jbang app install hawtio@hawtio/hawtio -Dhawtio.jbang.version=4.0.0.redhat-00029
```
and any parameters after the `<scriptRef>` would be inserted between `jbang run` and `<scriptRef> "$@"` in the installed script. But probably that might look a bit awkward for you.

Anyway, let me turn this question into a github issue.

**Reply by quintesse** [reply]
Thing is that the parameters after the `scriptRef` would better be used as default arguments to the script (added to the `"$@"` or perhaps instead of).

**Reply by tadayosi** [reply]
I see. Then, maybe another option would be something like:
```
jbang app install --run-params="-Dhawtio.jbang.version=4.0.0.redhat-00029" hawtio@hawtio/hawtio
```
as outlined at #1737.
